### PR TITLE
Capture :hover too when .is-active is used

### LINKED ISSUE #596: .{property}-{color}-on-active.is-active should override hover rules
That's a crazy long classname but for the hover state of active elements that have a different style than it's siblings:

on-hover-is-active

### PR DESCRIPTION
**Pull request by tristen**
- Fixes #596

### CODE REVIEW

File: scripts/build-color-variants.js [hunk 1]
```diff
@@ -480,11 +480,13 @@ function buildColorVariants(variables, config) {
       const colorValue = variables[color];
       return result += stripIndent(`
         .shadow-${color}-on-hover:hover,
-        .shadow-${color}-on-active.is-active {
+        .shadow-${color}-on-active.is-active,
+        .shadow-${color}-on-active.is-active:hover {
```
Comment (davidtheclark) [inline scripts/build-color-variants.js:484]: Is it not possible to accomplish the same thing by re-ordering? That could be more efficient.

Comment (tristen) [reply scripts/build-color-variants.js:484]: How would reordering help? `:hover` still wouldn't be captured.

Comment (davidtheclark) [reply scripts/build-color-variants.js:484]: `.x:hover` and `.x.y` have the same specificity. So whichever comes later in the stylesheet should override. You then don't need to create rules for `.x.y:hover`.

Comment (tristen) [reply scripts/build-color-variants.js:484]: 😭 I thought the specificity was slightly higher when targeting a pseudo selector. Sorry about that. will fix. 😭

Comment (tristen) [reply scripts/build-color-variants.js:484]: Hmm looks like they need to be decoupled from [being grouped together](https://github.com/mapbox/assembly/blob/573999d2bbdfd4f5ce9d4d2d95b9199cb96dfcd4/test/__snapshots__/build-color-variants.jest.js.snap#L9-L12) which means more lines.

Comment (tristen) [reply scripts/build-color-variants.js:484]: Discussed in chat with @davidtheclark: this looks like the better path to go down.

**Pull request merged**